schemas/rum/vital-schema.json

```diff
@@ -0,0 +1,56 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "rum/vital-schema.json",
+  "title": "RumVitalEvent",
+  "type": "object",
+  "description": "Schema of all properties of a Vital event",
+  "allOf": [
+    {
+      "$ref": "_common-schema.json"
+    },
+    {
+      "$ref": "_view-container-schema.json"
+    },
+    {
+      "required": ["type", "vital"],
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "RUM event type",
+          "const": "vital",
+          "readOnly": true
+        },
+        "vital": {
+          "type": "object",
+          "description": "Vital properties",
+          "required": ["id", "type"],
+          "properties": {
+            "type": {
+              "type": "string",
+              "description": "Type of the vital",
+              "enum": ["duration"],
+              "readOnly": true
+            },
+            "id": {
+              "type": "string",
+              "description": "UUID of the vital",
+              "pattern": "^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$",
+              "readOnly": true
+            },
+            "custom": {
+              "type": "object",
+              "description": "User custom vital. As vital name is used as facet path, it must contain only letters, digits, or the characters - _ . @ $",
```

Comment:
**❓ question:** ‏ ‏if it contains `.`, the field will be nested in an object (as `custom['foo.bar'] = 42` is equivalent to `custom.foo = { bar: 42 }` once processed). Is that intented?

If it is, we probably should adapt the type to something like:
```ts
custom: CustomVitalValue

type CustomVitalValue = number | Record<string, CustomVitalValue>
```
so we can either set it with a `.` or a nested object (like other Context)

Reply:
I copy-pasted the description from `view.custom_timings` without thinking too much about it.

From a SDK / ingestion perspective, the type should only be `number`.
From the app perspective, it will indeed be retrieved an an object but from a quick test, a `foo.bar` custom timing seems to be handled properly.

Since the types documented here are the ones sent by the SDK, I would tend to leave it as it is.
For the app team, it worth noting that the `.` should be handled like it is for `custom_timings`, cc @anediaz. 

wdyt?

Reply:
Ok. To me, it doesn't make sense to allow one way but not the other to send nested properties. I'm not sure we need to be similar to custom timings in this case. But it might be fine as long as other involved teams are aware of this.

Reply:
Discussed IRL, allowing object there does not seem to address any use case and it was surely unintentional for custom timings.
Preventing the use of `.` could avoid unnecessary logic to handle this case either on app or backend side.

wdyt @anediaz and @kbogtob?

Reply:
if I understand correctly, Browser SDK will prevent sending nested properties, and this will simplify treatments in the frontend and the backend right?

Reply:
yes, the idea would to be to not let customers use `.` for vital name in order to avoid the handle the object conversion done by the backend.

Reply:
I'll go ahead and merge this PR to not block depending tasks and move this discussion to a RFC.